### PR TITLE
Testing Enumareble methods with Rspec

### PR DESCRIPTION
**Pull request by yokoworks**
- Add specs for all Enumerable methods I previously created
- Wrap specs for each method in describe block, hints
- Instead of using before_each to define a variable for the test cases, use let
- Use one it block for each expected behaviour

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+  let(:bk_condit_num) { proc { |i| i.even? } }
+  let(:bk_condit_str) { proc { |item| item.is_a?(String) } }
+  let(:bk_condit_hash) { proc { |_key, value| value < 3 } }
+  let(:block_inject_num) { proc { |accum, i| accum * i } }
+  let(:block_inject_str) { proc { |accum, word| accum.size > word.size ? accum : word } }
+
+  describe '#my_each' do
+    it 'returns the same result as each method to num_array' do
+      expect(num_array.my_each(&block_for_num)).to eql(num_array.each(&block_for_num))
+    end
```
Comment (tirthajyoti-ghosh) [inline spec/enumerable_spec.rb:25]: `#each` **returns** the enumerable it's called on. In this test, you are basically checking if `#my_each` returns

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+      expect(str_array.my_each(&block_for_str)).to eql(str_array.each(&block_for_str))
+    end
+
+    it 'returns the same result as each method to range' do
+      expect(range.my_each(&block_for_num)).to eql(range.each(&block_for_num))
+    end
+
+    it 'returns the same result as each method to hash' do
+      expect(hash.my_each(&block_for_hash)).to eql(hash.each(&block_for_hash))
+    end
```
Comment (tirthajyoti-ghosh) [inline spec/enumerable_spec.rb:37]: These tests are not really checking the logic inside the block i.e., what is happening to the enumerable after `#my_each` applies the block in each iteration. You should be checking that. Right now, you are basically testing the equality of the return values of  `my_each` and `#each`, regardless of `hash` or `range` or `array`. 

Check this piece of code:

image

And then check this:

image

The above code will always print true because they both **return** the enumerable regardless of what is happening inside the block.

- [ ] Please add tests for the logic inside the block.

_Please note that this is the one place I have pointed this out. Please also add similar tests for the methods that **returns** enumerables._

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## What a remarkable project you have built 💯! This is impressive 🤩👌🧨!

The tests are very extensive. You are covering almost all of the edge cases.

### But in order to get it approved, we need to work on some changes. Let's see...

I have added some comments below. Please check them out.

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)
